packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import Arrange from 'terra-arrange';
+import styles from './SectionHeader.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+};
+
+const propTypes = {
+  /**
+   * Text to be displayed on the SectionHeader.
+   */
+  title: PropTypes.string.isRequired,
+  /**
+   * Callback function triggered when the accordion icon is clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Whether the accordion icon should be displayed in its 'open' or 'closed' position.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=2`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+};
+
+const defaultProps = {
+  onClick: undefined,
+  isOpen: false,
+  level: 2,
+};
+
+const isRecognizedKeyPress = event => ((event.nativeEvent.keyCode === KEYCODES.ENTER) || (event.nativeEvent.keyCode === KEYCODES.SPACE));
+
+class SectionHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { isActive: false };
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.wrapOnKeyUp = this.wrapOnKeyUp.bind(this);
+  }
+
+  wrapOnKeyDown(onKeyDown) {
+    return ((event) => {
+      if (isRecognizedKeyPress(event) && !this.state.isActive) {
+        this.setState({ isActive: true });
+
+        // Call the same function the user provides for a click event
+        if (this.props.onClick) {
+          this.props.onClick(event);
+        }
+      }
+
+      if (onKeyDown) {
+        onKeyDown(event);
+      }
+    });
+  }
+
+  wrapOnKeyUp(onKeyUp) {
+    return ((event) => {
+      if (isRecognizedKeyPress(event) && this.state.isActive) {
+        this.setState({ isActive: false });
```

Comment:
Didn't see this behavior in the design. Doesn't feel like keyUp should lose the focus on the header.

Reply:
It's not, it's losing its `active` state

Reply:
Sorry, that's what I meant actually. Do we have old behavior or docs we can reference? Just want to double check.

Reply:
No worries! We're currently doing this in [Button.jsx](https://github.com/cerner/terra-core/blob/Section-header/packages/terra-button/src/Button.jsx#L147)

Reply:
@JakeLaCombe This changes active state on keyUp only for `enter` or `spacebar` keys, other keys  won't trigger the active state change. `enter` and `spacebar` are used to toggle state as they are the keys used to select / interact with/ "click" native button and link elements.